### PR TITLE
Forwarder contracts

### PR DESCRIPTION
**Pull request by livingrock7**
- per remediations discussion (approved)
- idle permit and execute edge case 
- some test case updates

### CODE REVIEW

File: scripts/gas-price/get-gas-price.js [hunk 1]
```diff
@@ -33,4 +34,38 @@ function estimateGasPrice () {
       });
   });
 };
-module.exports = { estimateGasPrice };
+
+function estimateGasPriceMatic () {
+  if (etherscanApi) {
```
Comment (tomarsachin2271) [inline scripts/gas-price/get-gas-price.js:39]: this should be maticGasStationApi

Comment (livingrock7) [reply scripts/gas-price/get-gas-price.js:39]: updating now
